02.calendar/cal.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+require 'optparse'
+require 'date'
+
+options = ARGV.getopts('y:', 'm:')
+
+input_year = options["y"]
+input_month = options["m"]
+
+target_year = input_year.nil? ? Date.today.year : input_year.to_i
+target_month = input_month.nil? ? Date.today.month : input_month.to_i
+
+first_date = Date.new(target_year, target_month, 1)
+last_date = Date.new(target_year, target_month, -1)
+
+display_month = first_date.strftime('%B')
+display_days = 'Su Mo Tu We Th Fr Sa'
```

Comment:
```suggestion
month = first_date.strftime('%B')
days = 'Su Mo Tu We Th Fr Sa'
```

これぐらいの名前の方がシンプルでいいかも
https://bootcamp.fjord.jp/pages/avoid-too-specific-var-name

また、display_x だと「xを表示する」というメソッドに見えてしまうデメリットもあります。
https://qiita.com/jnchito/items/459d58ba652bf4763820